modules/dhcp/providers/server/isc.rb

```diff
@@ -78,13 +78,10 @@ def parse_config_and_leases_for_records
             opts.merge!(parse_record_options(data))
           end
         end
-        begin
-          subnet = @service.find_subnet(opts[:ip])
-          next unless subnet
-          ret_val << Proxy::DHCP::Reservation.new(opts.merge(:subnet => subnet))
-        rescue Exception => e
-          logger.warn "skipped #{hostname} - #{e}"
-        end
+        next if !opts.has_key?(:ip)
```

Comment:
I don't think we should skip the entry here, instead we probably need to skip the find_subnet call and then create a reservation with no subnet assigned so it can be deleted.

Looking back at the behaviour of 5f7fd34, it would delete the entry from the in memory store: https://github.com/theforeman/smart-proxy/blob/5f7fd34c7b4dc9b373a5fc1faeef4f21bcbc7919/modules/dhcp/providers/server/isc.rb#L68-L77

If we apply this as-is, we'll get entries in the store (which show up on GET /dhcp/:subnet) and we'll also need your other PR to fix deletion of entries that are actually non-existent.